web/web.go

```diff
@@ -240,9 +241,11 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
-	if err != nil {
-		panic(err)
+	if hostname == "" {
+		hostname, err = os.Hostname()
+		if err != nil {
+			panic(err)
+		}
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
```

Comment:
It will scrape localhost then, and it will also show that as localhost on the web UI, but the linkified version of the target URL will link to the configured hostname.